flang/lib/Lower/IntrinsicCall.cpp

```diff
@@ -1115,6 +1117,15 @@ mlir::Value IntrinsicLibrary::genDim(mlir::Type resultType,
   return builder.create<mlir::SelectOp>(loc, cmp, diff, zero);
 }
 
+// DPROD
+mlir::Value IntrinsicLibrary::genDprod(mlir::Type resultType,
+				       llvm::ArrayRef<mlir::Value> args) {
+  assert(args.size() == 2);
+  assert(fir::isa_real(resultType) && (resultType.getKind() == 8) && "Result must be double precision in DPROD");
+  auto res = builder.create<mlir::MulFOp>(loc, args[0], args[1]);
```

Comment:
The difference is subtle, but you should use a `fir::MulfOp` here. `mlir::MulFOp` does not support fir floating point types (we are using mlir float type when we can, but they are missing some float types required for flang). I agree it is rather unlikely that it is a problem in DPROD, but just to be safe I would stick to `fir::MulfOp`.

Reply:
Thanks for the feedback, I'll make the requested change.